### ISSUE TITLE
Error on Magento Cloud environment

### ISSUE THREAD

**Issue by francescopis** [algolia/algoliasearch-magento-2#801]
Hi,
on system.log i have this error
 report.CRITICAL: Can't create directory /app/my-platform-code/generated/code/Algolia/AlgoliaSearch/ViewModel/Adminhtml/Common/.
Class Algolia\AlgoliaSearch\ViewModel\Adminhtml\Common\Interceptor generation error: The requested class did not generate properly, because the 'generated' directory permission is read-only. If --- after running the 'bin/magento setup:di:compile' CLI command when the 'generated' directory permission is set to write --- the requested class did not generate properly, then you must add the generated class object to the signature of the related construct method, only. [] []

On magento cloud we can't change directory permission

**Comment by kalvis-o** [comment]
Hello,

I'm having the same issue on magento cloud environment.
The issue started appearing after algolia/algoliasearch-magento-2 module update from 1.7.2 to 1.11.3. 
Before module update, there were no such errors.

`report.CRITICAL: Can't create directory /app/generated/code/Algolia/AlgoliaSearch/ViewModel/Adminhtml/Common/.
Class Algolia\AlgoliaSearch\ViewModel\Adminhtml\Common\Interceptor generation error: The requested class did not generate properly, because the 'generated' directory permission is read-only. If --- after running the 'bin/magento setup:di:compile' CLI command when the 'generated' directory permission is set to write --- the requested class did not generate properly, then you must add the generated class object to the signature of the related construct method, only. [] []
`

**Comment by kalvis-o** [comment]
Update.

A temporary solution would be to write a plugin for one of the methods in the Common class to generate the interceptor. It will get rid of the error.

Example code:
```
class Common
{
    /**
     * Temporary fix to generate class interceptor
     * @param BaseCommon $subject
     * @param $section
     * @return null
     */
    public function beforeGetLinksConfig(BaseCommon $subject, $section) // @codingStandardsIgnoreLine UnusedFunctionParameter
    {
        return null;
    }
}
```

**Comment by bsuravech** [comment]
Hi everyone,

Thanks for reporting this and bringing this to our attention. I've created an internal ticket to resolve this issue so that this class is generated properly. We will update this issue once a PR is created. 

Thanks!

**Comment by bodinsamuel** [comment]
Hello team ☺️ 
A customer reached out to us to know more about the ETA ? is there any progress on this?

Thanks

**Comment by damcou** [comment]
Hello there, this issue has now been fixed by the latest release (1.12.1) .
Feel free to update your extension ;).

Best.
